tempto-runner/src/main/java/com/teradata/tempto/runner/TemptoRunner.java

```diff
@@ -0,0 +1,109 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.teradata.tempto.runner;
+
+import com.google.common.base.Joiner;
+import com.teradata.tempto.internal.configuration.TestConfigurationFactory;
+import com.teradata.tempto.internal.listeners.TestMetadataReader;
+import com.teradata.tempto.internal.listeners.TestNameGroupNameMethodSelector;
+import jdk.nashorn.internal.runtime.options.Option;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.TestNG;
+import org.testng.xml.XmlClass;
+import org.testng.xml.XmlPackage;
+import org.testng.xml.XmlSuite;
+import org.testng.xml.XmlTest;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static java.util.Collections.singletonList;
+
+public class TemptoRunner
+{
+    private static final Logger LOG = LoggerFactory.getLogger(TemptoRunner.class);
+    private final TemptoRunnerCommandLineParser parser;
+    private final TemptoRunnerOptions options;
+
+    private TemptoRunner(TemptoRunnerCommandLineParser parser, TemptoRunnerOptions options) {
+
+        this.parser = parser;
+        this.options = options;
+    }
+
+    public static void runTempto(TemptoRunnerCommandLineParser parser, TemptoRunnerOptions options)
+    {
+        new TemptoRunner(parser, options).run();
+    }
+
+    private void run()
+    {
+        LOG.debug("running tempto with options: {}", options);
+        if (options.helpRequested()) {
+            parser.printHelpMessage();
+            return;
+        }
+
+        XmlSuite testSuite = getXmlSuite();
+        setupTestsConfiguration();
+        TestNG testNG = new TestNG();
+        testNG.setXmlSuites(singletonList(testSuite));
+        setupTestsFiltering(testNG);
+        testNG.run();
+    }
+
+    private void setupTestsConfiguration()
+    {
+        System.setProperty(TestConfigurationFactory.TEST_CONFIGURATION_URI_KEY, options.getConfigFile());
+        System.setProperty(TestConfigurationFactory.LOCAL_TEST_CONFIGURATION_URI_KEY, options.getConfigFileLocal());
+    }
+
+    private void setupTestsFiltering(TestNG testNG)
+    {
+        if (!options.getTestGroups().isEmpty()) {
+            System.setProperty(TestNameGroupNameMethodSelector.TEST_GROUPS_TO_RUN_PROPERTY, Joiner.on(',').join(options.getTestGroups()));
+        }
+        if (!options.getExcludeGroups().isEmpty()) {
+            System.setProperty(TestNameGroupNameMethodSelector.TEST_GROUPS_TO_EXCLUDE_PROPERTY, Joiner.on(',').join(options.getExcludeGroups()));
+        }
+        if (!options.getTests().isEmpty()) {
+            System.setProperty(TestNameGroupNameMethodSelector.TEST_NAMES_TO_RUN_PROPERTY, Joiner.on(',').join(options.getTests()));
+        }
+        testNG.addMethodSelector("com.teradata.tempto.internal.listeners.TestNameGroupNameMethodSelector", 20);
```

Comment:
also, don't use string class name, use TestNameGroupNameMethodSelector Class object and get class name from it.